### PR TITLE
Fmod studio update

### PR DESCRIPTION
**Pull request by addictgamer**
All of the merge conflicts

TODO: For MSVC, can we have the install targets also copy over all of the needed DLLs?
TODO: double-check OpenAL

### CODE REVIEW

File: src/eos_editor.cpp [hunk 1]
```diff
@@ -0,0 +1,23 @@
+/*
```
Comment (addictgamer) [inline src/eos_editor.cpp:1]: Ended up not really using this. It wasn't working 100%, probably needs a few more things...oh well. Something we can explore later.

File: src/init_game.cpp [hunk 2]
```diff
@@ -861,108 +708,109 @@ void deinitGame()
 		safePacketsReceivedMap[c].clear();
 	}
 #ifdef SOUND
-#ifdef USE_OPENAL
+#ifdef USE_OPENAL //TODO: OpenAL is now all of the broken...
```
Comment (addictgamer) [inline src/init_game.cpp:711]: Not sure if we can just reuse the hack below...might need to rearchitect this area.

File: src/engine/audio/music.cpp [hunk 3]
```diff
@@ -0,0 +1,656 @@
...
+		playMusic(sanctummusic, true, true, true);
+		levelmusicplaying = false;
+		devilmusicplaying = false;
+		herxmusicplaying = false;
+		minotaurmusicplaying = false;
+		combatmusicplaying = false;
+		sanctummusicplaying = true;
+		shopmusicplaying = false;
+		fadein_increment = default_fadein_increment * 2;
+		fadeout_increment = default_fadeout_increment * 2;
```
Comment (addictgamer) [inline src/engine/audio/music.cpp:594]: ...we really should refactor these into something like
```
clearMusicStates();
sanctummusicplaying = true;
```

File: src/engine/audio/sound.cpp [hunk 4]
```diff
@@ -126,47 +68,26 @@ void sound_update()
...
 	up.x = 0;
 	up.y = 1;
 	up.z = 0;
-	/*up.x = rx;
-	up.y = ry;
-	up.z = rz;*/
 
 	//FMOD_System_Set3DListenerAttributes(fmod_system, 0, &position, &velocity, &forward, &up);
-	FMOD_System_Set3DListenerAttributes(fmod_system, 0, &position, 0, &forward, &up);
+	fmod_system->set3DListenerAttributes(0, &position, 0, &forward, &up); //TODO: For splitscreen, change the listener number to the current player...
```
Comment (addictgamer) [inline src/engine/audio/sound.cpp:80]: @SheridanR FYI

File: src/engine/audio/sound.cpp [hunk 5]
```diff
@@ -1297,7 +1162,7 @@ void physfsReloadMusic(bool &introMusicChanged, bool reloadAll)
 
 	int index = 0;
 #ifdef USE_OPENAL
-#define FMOD_System_CreateStream(A, B, C, D, E) OPENAL_CreateStreamSound(B, E)
+#define FMOD_System_CreateStream(A, B, C, D, E) OPENAL_CreateStreamSound(B, E) //TODO: If this is still needed, it's probably now broke!
```
Comment (addictgamer) [inline src/engine/audio/sound.cpp:1165]: Another place that the OpenAL code will need special attention...

File: src/engine/audio/sound.cpp [hunk 6]
```diff
@@ -1319,317 +1184,188 @@ void physfsReloadMusic(bool &introMusicChanged, bool reloadAll)
...
-						fmod_result = FMOD_System_CreateStream(fmod_system, musicDir.c_str(), FMOD_SOFTWARE, NULL, &tutorialmusic);
+						fmod_result = fmod_system->createStream(musicDir.c_str(), FMOD_2D, nullptr, &tutorialmusic);
 						break;
 					default:
 						break;
 				}
+				if ( FMODErrorCheck() )
+				{
+					printlog("[PhysFS]: ERROR: Failed reloading music file \"%s\".", filename.c_str());
+					//TODO: Handle error? Abort? Fling pies at people?
```
Comment (addictgamer) [inline src/engine/audio/sound.cpp:1323]: Most definitely fling pies at people.

...mmmm, I could go for some pie.

File: src/engine/audio/sound.cpp (hunk 4 above)
Comment (SheridanR) [reply src/engine/audio/sound.cpp:80]: Thanks for pointing this out, I made a personal note about it

**Review by SheridanR — APPROVED** [review]
This looks good to me. Too bad OpenAL is basically broken now, but we probably don't have a reason to maintain it for the time being. Nice work :)

**Comment by addictgamer** [comment]
> This looks good to me. Too bad OpenAL is basically broken now, but we probably don't have a reason to maintain it for the time being. Nice work :)

Ya...I'll look back at OpenAL once we close on some of the other P1s on my plate. I already have some ideas in mind on how we can address it.

**Pull request merged**